shared/modules/persistence_module.go

```diff
@@ -33,6 +38,22 @@ type PersistenceModule interface {
 	HandleDebugMessage(*messaging.DebugMessage) error
 }
 
+// TreeStore is fulfilled by the treeStore to create an
+// atomic tree component for use by the peristence context.
+type TreeStore interface {
+	// Update returns the new state hash for a given height.
+	// * Height is passed through to the Update function
+	// and used by the queries against the TxIndexer and the
```

Comment:
Optional NIT: I feel like single lining this whole statement (even if it's > 80 chars) will make it easier to read